src/main/java/domain/Line.java

```diff
@@ -1,33 +1,37 @@
 package domain;
 
-import utils.BooleanGenerator;
-
 import java.util.ArrayList;
 import java.util.List;
+import utils.StepGenerator;
 
 public class Line {
 
-    private final List<Boolean> points = new ArrayList<>();
+    private final List<StepPoint> stepPoints = new ArrayList<>();
 
-    public Line(int count, BooleanGenerator booleanGenerator) {
-        points.add(booleanGenerator.generate());
-        for (int index = 1; index < count; index++) {
-            points.add(makeOnePoint(index, booleanGenerator));
+    public Line(int numberOfCell, StepGenerator stepGenerator) {
+        stepPoints.add(stepGenerator.generate());
+        for (int cellIndex = 1; cellIndex < numberOfCell; cellIndex++) {
```

Comment:
for문이 1로 시작하는 게 눈에 띄네요! (isPreviousStepExist에서 index 관련 예외를 막기 위함이죠?)

흔하지 않은 경우여서 다른 개발자가 봤을 때 인지적 혼란을 일으킬 수 있는 코드예요. 왜 1인지 추측해야 합니다.
개발자의 업무 중 70% 이상이 "코드를 읽는 것"이라는 분석도 있는데요, 그래서 가독성이 중요합니다 (다른 개발자의 업무 효율을 올려줌)

그래서 저라면 주석을 통해 표현하거나, 우테코가 좋아하는 상수화로 이 매직넘버를 처리합니다.
```suggestion
        // 이전 step point를 비교해야해서 1로 시작
        for (int cellIndex = 1; cellIndex < numberOfCell; cellIndex++) {
```

```suggestion
        for (int cellIndex = EXCEPT_FIRST_INDEX; cellIndex < numberOfCell; cellIndex++) {
```
상수화 방식은 1로 하나 EX..로 하나 뭐가 다른가 싶기도 하겠지만 실수가 아니라 의도가 있는걸 전달하는데 가치가 있다고 생각합니다 (실수인지 아닌지 판단하려면 코드 전체를 봐야합니다)

저는 영어 단어로 이 상황을 알기 좋게 표현할 자신이 없어서 실무였으면 그냥 주석을 썼습니다